### PR TITLE
feat(SharedClusters): add database_name to assignment resource

### PR DESCRIPTION
**Pull request by asms**
Add the new field `database_name` to the PostgresqlClusterServiceAssignment resource.

[QSS-2346]

[QSS-2346]: https://outreach-io.atlassian.net/browse/QSS-2346?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kubernetes/database.libsonnet [hunk 1]
```diff
@@ -37,6 +37,8 @@ local k = import 'kubernetes/kube.libsonnet';
     local this = self,
     // bento is the bento which contains the database cluster and application
     bento:: error 'bento is required',
+    // database_name is the name of the application database to create within the cluster
```
Comment (AndrewWinterman) [inline kubernetes/database.libsonnet:40]: I think it's helpful to try to describe the namespace win this comment, e.g.:

```suggestion
    // database_name is the name of the database to create within the postgresql cluster. Schemas for the application will be created in the provided datatabase. See https://www.postgresql.org/docs/current/ddl-schemas.html
```

File: kubernetes/database.libsonnet [hunk 2]
```diff
@@ -37,6 +37,8 @@ local k = import 'kubernetes/kube.libsonnet';
...
+    database_name:: error 'database_name is required',
```
Comment (AndrewWinterman) [inline kubernetes/database.libsonnet:41]: is marking this required going to cause issues for folks?

**Review by AndrewWinterman — APPROVED** [review]
no blockers from me.

File: kubernetes/database.libsonnet (hunk 2 above)
Comment (asms) [reply kubernetes/database.libsonnet:41]: It should be fine. This is one the `PostgresqlClusterServiceAssignment`, which isn't used yet. You were probably thinking this was the `PostgresqlDatabaseCluster`.

**Review by jan-ludvik — APPROVED** [review]

**Review by dvagapov — APPROVED** [review]

**Pull request merged**